common/app/navigation/helpers/UrlHelpers.scala

```diff
@@ -53,6 +56,11 @@ object UrlHelpers {
     NavLink("Subscribe", getReaderRevenueUrl(SupportSubscribe, SideMenu), classList = Seq("js-subscribe"))
   )
 
+  private val uriEncoder = UriConfig.default.copy(
+    // The default encoder is not enough for json in URL query strings
+    queryEncoder = PercentEncoder(PercentEncoder.QUERY_CHARS_TO_ENCODE ++ Set('"', ':', ',') )
```

Comment:
are you sure, according to the [RFC](https://tools.ietf.org/html/rfc3986#section-3.4), colon and comma are fine:
query       = *( pchar / "/" / "?" )
  pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
   sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
                 / "*" / "+" / "," / ";" / "="

Only DQUOTE is not allowed as far as I can tell.

Having said that, overescaping isn't a problem, we could escape every single character if we wanted!  HTML forms indeed do a lot more escaping than the URI standard requires.

Reply:
My interpretation of the RFC is different. Since `:` is a reserved character (specifically one of the `gen-delims` https://tools.ietf.org/html/rfc3986#section-2.2) it does need to be encoded if it's within a query string parameter. Obviously characters in the reserved set are allowed in URLs, but they have special meaning - e.g. if you don't percent-encode `&` within a query string parameter value, it will be interpreted as the start of another parameter.

This is roughly the difference between https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI

That said, strictly speaking that would mean we should apply this escaping rule just to the param values, not the whole URL. Though since we know the types of links we're generating here, it's probably safe to assume that `:` and `,` are not used as delimiters.

Reply:
oh I see you've addressed that last point below